manager/raftpicker/raftpicker.go

```diff
@@ -98,14 +104,16 @@ type ConnSelector struct {
 	cc      *grpc.ClientConn
 	cluster RaftCluster
 	opts    []grpc.DialOption
+	ctx     context.Context
 }
 
 // NewConnSelector returns new ConnSelector with cluster and grpc.DialOpts which
 // will be used for Dial on first call of Conn.
-func NewConnSelector(cluster RaftCluster, opts ...grpc.DialOption) *ConnSelector {
+func NewConnSelector(ctx context.Context, cluster RaftCluster, opts ...grpc.DialOption) *ConnSelector {
 	return &ConnSelector{
 		cluster: cluster,
 		opts:    opts,
+		ctx:     ctx,
```

Comment:
If we're going to have a context field here, might be a good idea to allocate a cancelled context and then have the cancel function get called on close.